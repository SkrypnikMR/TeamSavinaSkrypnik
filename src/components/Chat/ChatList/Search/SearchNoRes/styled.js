import styled from 'styled-components';
import { colorDefault } from '/src/components/UI/baseLayout';

export const StNoResWrapper = styled.div`
    display: flex;
    justify-content: center; 
    align-items: center;
    flex-direction: column;
`;
export const StNoResTitle = styled.p`
  color: ${({ color = colorDefault }) => color};
  font-family: 'Play', sans-serif;
`;
export const StNoResText = styled.span`
  color: ${({ color = colorDefault }) => color};
  font-family: 'Play', sans-serif;
`;
