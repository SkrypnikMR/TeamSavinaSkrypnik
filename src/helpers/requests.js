import { routes } from '../constants/routes';

const { baseUrl } = routes;

export const getRequest = async (url, optionalHeaders = {}) => {
  const headers = Object.assign({}, optionalHeaders);
  const options = { method: 'GET', headers };
  const answer = await fetch(`${baseUrl}${url}`, options);
  if (!answer.ok) throw new Error(await answer.json());
  const result = await answer.json();
  return result;
};

export const postRequest = async (url, body, optionalHeaders = {}) => {
  const headers = Object.assign({ 'Content-Type': 'application/json' }, optionalHeaders);
  const options = {
    method: 'POST',
    headers,
    body: JSON.stringify(body),
  };
  const answer = await fetch(`${baseUrl}${url}`, options);
  return answer;
};

export const putRequest = async (url, body, optionalHeaders = {}) => {
  const headers = Object.assign({ 'Content-Type': 'application/json' }, optionalHeaders);
  const options = {
    method: 'PUT',
    headers,
    body: JSON.stringify(body),
  };
  const answer = await fetch(`${baseUrl}${url}`, options);
  const result = await answer.json();
  return result;
};
export const createAuthHeader = (support) => { return { Authorization: support.getCookie('token') }; };
