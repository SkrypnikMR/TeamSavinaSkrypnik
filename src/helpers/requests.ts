import { routes } from '../constants/routes';
import { TUrl, TBody } from './types/requestTypes';

const { baseUrl } = routes;

export const getRequest = async (url: TUrl) => {
  const options = { method: 'GET' };
  const answer = await fetch(`${baseUrl}${url}`, options);
  return answer;
};

export const postRequest = async (url : TUrl, body : TBody) : Promise<Response> => {
      const options = {
              method: 'POST',
              body: JSON.stringify(body),
              headers: { 'Content-Type': 'application/json' },
      };
  const answer = await fetch(`${baseUrl}${url}`, options);
  return answer;
};
