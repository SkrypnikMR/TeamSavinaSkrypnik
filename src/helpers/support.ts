import moment from 'moment';

export const support = {
    setSessionStorageItem: (name, data) => {
        if (typeof data !== 'string') data = JSON.stringify(data);
        localStorage.setItem(name, data);
    },
    killSessionStorageItem: (name) => {
        localStorage.removeItem(name);
    },
    getSessionStorageItem: (name) => {
        const item = localStorage.getItem(name);
        return item?.charAt(0) === '{' ? JSON.parse(item) : item;
    },
    getMessagesFolders: (rooms) => {
        const messagesFolders = {};
        rooms.forEach((room) => {
            const { room_name } = room;
            messagesFolders[room_name] = [];
        });
        return messagesFolders;
    },
    getFormatedDate: date => moment(date).format('YYYY-MM-DD HH:mm:ss'),
    playAudio: (path) => {
        const audio = new Audio(path);
        audio.play();
    },
};
