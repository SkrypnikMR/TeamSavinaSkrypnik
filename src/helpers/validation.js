import { regExp } from '../constants/regExp';

export const validation = {
    registrationValidation: ({ email, password, confirm, firstName, lastName }) => {
        if (!regExp.emailRegExp.test(email) || !email) return { message: 'invalid_email', isValid: false };
        if (!regExp.passwordRegExp.test(password) || !password) return { message: 'invalid_password', isValid: false };
        if (password !== confirm) return { message: 'password_mismatch', isValid: false };
        if (!regExp.firstAndLastNameRegExp.test(firstName) || !firstName) return { message: 'invalid_first_name', isValid: false };
        if (!regExp.firstAndLastNameRegExp.test(lastName) || !lastName) return { message: 'invalid_last_name', isValid: false };
        return { message: '', isValid: true };
    },
    loginValidation: ({ email, password }) => {
        if (!regExp.emailRegExp.test(email) || !email) return { message: 'invalid_email', isValid: false };
        if (!regExp.passwordRegExp.test(password) || !password) return { message: 'invalid_password', isValid: false };
        return { message: '', isValid: true };
    },
    numberValidation: (age) => Number.isNaN(Number(age)),
    inputMyAccountValidation: ({ firstName, lastName }) => {
        if (!regExp.firstAndLastNameRegExp.test(firstName) || !firstName) return { message: 'invalid_first_name', isValid: false };
        if (!regExp.firstAndLastNameRegExp.test(lastName) || !lastName) return { message: 'invalid_last_name', isValid: false };
        return { message: '', isValid: true };
    },
};
