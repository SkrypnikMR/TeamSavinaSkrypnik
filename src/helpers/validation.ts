import { TRegValidation, TLoginValidation, TAge } from './types/validationTypes';
import { regExp } from '../constants/regExp';

export const validation = {
    registrationValidation: ({ login, password, confirm } : TRegValidation) => {
        if (!regExp.loginRegExp.test(login) || !login) return { message: 'invalid_login', isValid: false };
        if (!regExp.passwordRegExp.test(password) || !password) return { message: 'invalid_password', isValid: false };
        if (password !== confirm) return { message: 'password_mismatch', isValid: false };
        return { message: '', isValid: true };
    },
    loginValidation: ({ login, password } : TLoginValidation) => {
        if (!regExp.loginRegExp.test(login) || !login) return { message: 'invalid_login', isValid: false };
        if (!regExp.passwordRegExp.test(password) || !password) return { message: 'invalid_password', isValid: false };
        return { message: '', isValid: true };
    },
    numberValidation: (age: TAge) => Number.isNaN(Number(age)),
};
