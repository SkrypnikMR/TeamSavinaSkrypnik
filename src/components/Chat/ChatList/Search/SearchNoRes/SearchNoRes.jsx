import React from 'react';
import { useTranslation } from 'react-i18next';
import { StNoResTitle, StNoResWrapper, StNoResText } from './styled';

const SearchNoRes = () => {
    const { t } = useTranslation();
    return (
        <StNoResWrapper>
            <StNoResTitle>{t('no_result')}</StNoResTitle>
            <StNoResText>{t('try_new')}</StNoResText>
        </StNoResWrapper>
    );
};

export default SearchNoRes;
