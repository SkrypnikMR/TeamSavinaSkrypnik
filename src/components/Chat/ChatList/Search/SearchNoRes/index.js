import SearchNoRes from './SearchNoRes.jsx';

export default SearchNoRes;
