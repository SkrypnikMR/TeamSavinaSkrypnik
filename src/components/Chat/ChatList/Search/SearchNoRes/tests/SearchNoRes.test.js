import React from 'react';
import { shallowSmart, mountSmart } from '/src/helpers/testHelper';
import SearchNoRes from '../SearchNoRes';

describe('SearchNoRes', () => {
    it('Should match snapshot', () => {
        const component = shallowSmart(<SearchNoRes />);
        expect(component.html).toMatchSnapshot();
    });
    it('Should render StNoResWrapper', () => {
        const component = mountSmart(<SearchNoRes />);
        expect(component.find('styled__StNoResWrapper')).toHaveLength(1);
    });
    it('Should render StNoResTitle', () => {
        const component = mountSmart(<SearchNoRes />);
        expect(component.find('styled__StNoResTitle')).toHaveLength(1);
    });
    it('Should render StNoResText', () => {
        const component = mountSmart(<SearchNoRes />);
        expect(component.find('styled__StNoResText')).toHaveLength(1);
    });
});
