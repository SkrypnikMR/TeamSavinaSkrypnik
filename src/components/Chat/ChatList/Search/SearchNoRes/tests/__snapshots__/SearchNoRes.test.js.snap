// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchNoRes Should match snapshot 1`] = `[Function]`;
